tests/test_polydata.py

```diff
@@ -268,7 +268,8 @@ def test_multi_ray_trace(sphere):
 @pytest.mark.skipif(not system_supports_plotting(), reason="Requires system to support plotting")
 def test_plot_curvature(sphere):
     cpos = sphere.plot_curvature(off_screen=True)
-    assert isinstance(cpos, pyvista.CameraPosition)
+    # cpos is None within unit testing
+    assert cpos is None
```

Comment:
I've only barely skimmed the PR yet so this might be answered elsewhere: is it specified when `cpos` is returned and when it isn't?

And is it not a bad idea to change API behaviour for testing? When we're no longer testing production code, tests start losing meaning.

If I understand correctly the goal is to make example code blocks pretty by suppressing `cpos` being returned. So can't we monkeypatch `plot`/`show` /etc. calls only for docs generation? This shouldn't affect unit tests.

Reply:
I honestly would love to remove returning `cpos` altogether and then add an optional `return_cpos` to our kwargs.  How about we do that instead.

I agree that it's bad practice to change outputs just for testing, which defeats testing...

Reply:
A new kwarg would work. Then again you've been working to make return values homogeneous for `inplace=True`, so this would then defeat _that_ a bit. But practicality beats purity so it could work.

Reply:
If you're fine with it for now, let's keep as is and fix in a follow-up PR.

Reply:
Well I'm not crazy about it, but if you promise to fix that soon in the follow-up then it won't be like this for long :P

Reply:
> Well I'm not crazy about it, but if you promise to fix that soon in the follow-up then it won't be like this for long :P

This PR is huge as is.  I'll make another PR correcting the behavior.